### PR TITLE
setup: Server NPM initialization

### PR DESCRIPTION
**Pull request by pejrak**

### CODE REVIEW

**Comment by pejrak** [comment]
please help with the test @jCobbSK

File: server/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "server",
+  "version": "1.0.0",
+  "description": "SUP server for web and API",
+  "main": "index.js",
+  "scripts": {
+    "test": "jest",
```
Comment (jCobbSK) [inline server/package.json]: Tests do not work me, it's complaining about ES modules.

File: server/src/http/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const app: express.Application = express();
+const corsMiddleware = cors({ origin: '*', preflightContinue: true });
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+app.use(logger('dev'));
+app.use(corsMiddleware);
+app.options('*', corsMiddleware);
+routes(app).listen(process.env.PORT);
```
Comment (jCobbSK) [inline server/src/http/index.ts]: We should define some default port and probably also console log that server is running on specific port.

File: server/src/http/index.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export default app;
```
Comment (jCobbSK) [inline server/src/http/index.ts]: Based on https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html , http://jypepin.com/anti-pattern-es6-default-exports and many more, default exports are considered harmful, maybe we should (maybe not in this PR) move away from it and prefer named exports.

File: server/package.json (hunk 1 above)
Comment (pejrak) [reply server/package.json]: should work now... had to add transformation for jest ts files

File: server/src/http/index.ts (hunk 2 above)
Comment (pejrak) [reply server/src/http/index.ts]: changed

Comment (pejrak) [reply server/src/http/index.ts]: added console write on listen start

File: server/src/http/index.ts (hunk 3 above)
Comment (pejrak) [reply server/src/http/index.ts]: done

**Pull request merged**